### PR TITLE
Handle Pathname object in Pathname constructor

### PR DESCRIPTION
**Pull request by Mogztter**

### CODE REVIEW

File: stdlib/pathname.rb [hunk 1]
```diff
@@ -1,7 +1,12 @@
...
+
   def initialize(path)
-    raise ArgumentError if path == "\0"
-    @path = path
+    #raise ArgumentError if path == "\0"
+    if Pathname === path
+      @path = path.path.to_s
+    else
+      @path = path.to_s
+    end
```
Comment (elia) [inline stdlib/pathname.rb]: maybe `raise ArgumentError if @path == "\0"` can be added here?

Comment (Mogztter) [inline stdlib/pathname.rb]: Correct

**Comment by Mogztter** [comment]
Rebased on master

**Comment by elia** [comment]
:+1: thanks!

**Pull request merged**